### DISCUSSION TITLE
TFT Quantile Forecast - Get results

### DISCUSSION THREAD

**Discussion by tackes** [awslabs/gluonts#1966] [Q&A] [answered]
The other models return a prediction results as gluonts.model.forecast.SampleForecast, but the TFT returns a quantileforecast.

How can we extract the forecast from the quantile forecast?

**Comment by lostella** [comment, marked as answer]
@tackes what do you mean exactly by "extract the forecast"? The idea is that there are multiple ways of representing a predicted distribution: samples (`SampleForecast`), a fixed set of quantiles (`QuantileForecast`) or a parametric distribution (say, Gaussian, student-t or negative binomial, using `DistributionForecast`). Many models return a `SampleForecast` because their inner logic produce sample paths of the predicted behavior for a series, but others (like TFT) are inherently trained to fit specific quantiles.

This said, if you are interested in extracting a specific quantile from the forecast, you should look into the [`quantile` method of QuantileForecast](https://github.com/awslabs/gluon-ts/blob/7d7dd2937251a22f6bb8b367845d0037ed4d73d8/src/gluonts/model/forecast.py#L530). E.g.:

```
p50_prediction = forecast.quantile(0.5)
```

**Comment by tackes** [comment]
Thank you Lostella - This is what I was looking for:
` forecast.quantile(0.5)`

That said I am confused by the output.  
In this image below, the forecast keys (block 08) are in order ['0.5','0.1','0.9'].   WHen i look at the values of the results, the first forecast array looks like it should be quantile(0.1) and the 2nd should be quantile(0.5), but it is switched.  (ie. the values in the 2nd array are between the 1st and 3rd hence should be the median value.)  When i print quantile(0.5), the first array is printed (-7.048, -6.765.. etc).


In block 67, I produced a forecast with only 1 output, ie. the quantile(0.5), and the values in that output is very similar to the middle array values in block 08, but for which the forecast keys (and print statement) say it is quantile(0.1)

Is this a bug or am I misunderstanding something?

![Capture](https://user-images.githubusercontent.com/9125316/167131659-949b63d3-4680-4636-a6a4-1097f9c5001b.PNG)

**Reply by lostella** [reply]
This appears to be an already solved bug, see https://github.com/awslabs/gluon-ts/pull/1952

The fix was released in version 0.9.4, so if you are running an older version you just need to update to the latest. Let me know if this works!
